eeps/eep-0064.md

```diff
@@ -0,0 +1,187 @@
+[EEP-62][],
+[PR-7343][]).
+
+Triple-quoted binary-strings do not allow string interpolation and, if we were
+to accept them, only calls to statically known values should be allowed, e.g.,
+macros. This design decision forbids the creation of dynamic documentation that
+generates code at runtime, e.g., doing an IO call to get a value not known
+statically and based on it perform an action (notice that this is allowed when
+using string interpolation).
+
+### Binary-String Design
+
+To make the usage of binary-string similar to other languages, this EEP proposal
+makes the binary-string design decisions similar to the ones for the Elixir language.
+
+#### Binary-String Start and End
+
+Binary-strings must start and end with triple-quotes in their own lines:
+
+*Code Example*
+
+    X = """
+        This is the beginning of the triple-quote text
+        """
+
+*Documentation Example*
+
+    -doc """
+         Removes double-quotes (") from a given string
+         """
+    remove_double_quotes(X) -> 
+
+#### Binary-Strings Closing of Triple-Quotes
+
+The closing of the triple-quotes denotes the space (indentation):
+
+*1. Code Example*
+
+    >> X = """
+           This text
+           has no indentation
+           """
+    
+    Equivalent to: <<"This text\nhas no indentation\n">>
+
+*2. Code Example*
+
+    >> Y = """
+           This text
+           has indentation
+    """
+    
+    Equivalent to: <<"       This text\n       has indentation\n">>
+
+*3. Code Example*
+
+    >> Z = """
+      This text has
+      two space indentation
+    """
+    
+    Equivalent to: <<"  This text has\n  two space indentation\n">>
+
+*1. Documentation Example - No indentation*
+
+    -doc """
+         Removes double-quotes (") from a given string
+         """
+    remove_double_quotes(X) -> 
+
+*2. Documentation Example - No indentation*
+
+    -doc """
+    Removes double-quotes (") from a given string
+    """
+    remove_double_quotes(X) -> 
+
+*3. Documentation Example - Indentation*
+
+    -doc """
+         Removes double-quotes (") from a given string
+    """
+    remove_double_quotes(X) -> 
+
+#### Binary-Strings Errors
+
+An error should be raised when there is text that starts before the closing of triple-quotes, where
+the error makes the code written using triple-quotes to have a uniform style:
+
+*Code Example*
+
+    >> Err = """
+    This shall be an error
+             """
+   
+    error             
+
+*Documentation Example*
+
+    -doc """
+    Removes double-quotes (") from a given string
+         """
+    remove_double_quotes(X) -> 
+
+#### Escaping Newlines in Triple-Quotes
+
+Escaping new lines using `\`:
+
+To write `\` inside triple quotes, one needs to use `\\`. [EEP-59][] should
+ignore the meaning of `\` if inside a code block.
+
+*Code Example*
+
+    >> X = """
+           This text \
+           has no indentation
+           """
+    
+    Equivalent to: <<"This text has no indentation\n">>
```

Comment:
Makes sense the possibility to comment after `\`?
For example:
```
"""
foo \ % Maybe I want to break this line to add a comment about it
bar
"""
```

Reply:
> I think that's ambiguous because `\` is no longer at the end. 

Oh, this is true and makes things tricky.

> Should `\%` be allowed then at the end? 

Oh, using `\%` makes sense to me.

> And, if so, how do you actually know `\` is at the end?

I think this is easy, I've implemented this now based on your idea of the `\%` on [this branch](https://github.com/williamthome/heredocs/tree/feat/comment) of a side project called `heredocs`. That's what the implementation outputs:
```
"foo bar" = """
            foo \% This is a comment and must be ignored
            bar\
            """
```

I don't know if this is a good idea, but it's possible.

Reply:
My concern is that the last example is still ambiguous with escaping the `%` character itself. :)

Reply:
I do not understand why you want a comment in a multi line string.
All in it should be string content, otherwise it isn't very useful, right?

We want it for Markdown documentation, and Markdown uses trailing `\` for forcing a line break, which clashes head on with using trailing `\` for newline escape.

So I do not know if we want newline escape.  Or any escape codes at all, for that matter, since Markdown uses `\` for escaping things, just as the regular Erlang string syntax, and these may also collide...

Reply:
> We want it for Markdown documentation, and Markdown uses trailing \ for forcing a line break, which clashes head on with using trailing \ for newline escape.

`\` will escape the new line character. To force the line break, you need `\\` (because Markdown needs to see the `\` itself). So to me it makes logical sense: if you use `\\`, you are escaping the character that would escape/hide the line break. :)

Reply:
Comment inside the string makes the syntax too complex IMO. It's already possible to put two strings side-by-side to concatenate them, e.g.

```Erlang
"foo " % this comment is ignored
"bar" = """
        foo \
        """
        %% this comment would be ignored too
        """
        bar\
        """.
```

Reply:
> We want it for Markdown documentation

Sorry @RaimoNiskanen, I commented without thinking in Markdown.
BTW, we can write comments in Markdown in [many different ways](https://stackoverflow.com/questions/4823468/comments-in-markdown).